### ISSUE TITLE
Integrating react-router in pure Flux

### ISSUE THREAD

**Issue by rybon** [rackt/react-router#2045]
I have a bit of trouble understanding how to integrate react-router in a 'pure' Flux implementation. What I mean by pure, is that every UI change is triggered by dispatching an action to the store. Looking at the react-router way of doing things, the router changes the UI for you by resolving and rendering the `RouteHandler` in the `run` callback. Although it is possible to dispatch the router state to the store from inside the `run` callback, I cannot figure out how I can use the resulting state to resolve and render the `RouteHandler` elsewhere. Say, from within the App component where I currently render `<RouteHandler  />`.

I consider this to be important, because it means one can assume the current UI state is a complete reflection of the store state, that by itself is the result of dispatching a sequence of actions. It looks like react-router breaks that contract, by rendering the RouteHandler for you.

Currently I'm using react-router 0.13 with Redux.

Basically, I want to go from this code:

```
Router.run(routes, function(RouteHandler) {
  React.render(<RouteHandler />, document.body);
});
```

To:

```
Router.run(routes, function(RouteHandler, state) {
  Store.dispatch({
    type: ROUTE_UPDATE,
    state: state
  });
});
React.render(<App />, document.body);

// in app.js, assuming App is subscribed to store changes
render: function() {
  return (
    <div>
      { ReactRouter.resolveRouteHandler(Store.getState().router) } // something like this
    </div>
}
```

This way, the UI can only change through store updates. I can capture a sequence of actions and replay them in the same order to reconstruct the same state and therefore UI.

How do I accomplish this with react-router?

**Comment by knowbody** [comment]
Have you read [the docs](https://github.com/rackt/react-router/blob/0.13.x/docs/guides/flux.md)?

**Comment by rybon** [comment]
Yes, I have read the docs. This problem is not addressed anywhere.

Suppose I have a sequence of actions:

```
[{ type: ROUTE_UPDATE, data: routerState }, { type: ROUTE_UPDATE, data: routerState }, { type: ROUTE_UPDATE, data: routerState }]
```

When I dispatch these actions to the Store, how can I resolve the correct RouteHandler based on this payload? I do not want to render the RouteHandler in an action creator, as I cannot replay these methods, only the actions they spawned. I cannot dispatch the RouteHandler to the Store, as it only stores primitives, objects and arrays.

That leaves only one option as far as I understand: resolving the RouteHandler based on the props from the Store inside the App component.

Thoughts?
